File: Restaurant/Java/Restaurant.java
package restaurant;

import java.io.*;
import java.util.*;

class Restaurant{
	
	String name;
	double sum;
	ArrayList<Drink> drinks;
	ArrayList<Entree> meals;
	ArrayList<Side> extras;
	ArrayList<Dessert> sweets;
	ArrayList<Item> order;
	
	Restaurant(){
	}
	
	Restaurant(String name){
		this.name = name;
		this.drinks = new ArrayList<Drink>();
		this.meals = new ArrayList<Entree>();
		this.extras = new ArrayList<Side>();
		this.sweets = new ArrayList<Dessert>();
		this.order = new ArrayList<Item>();
		
	}
	
	public String getName() {
		return this.name;
	}
	
	public void addDrink(Drink d) {
		this.drinks.add(d);
	}
	
	public void addMeal(Entree m) {
		this.meals.add(m);
	}
	
	public void addSide(Side e) {
		this.extras.add(e);
	}
	
	public void addDessert(Dessert s) {
		this.sweets.add(s);
	}
	
	public void addToOrder(Item food) {
		this.order.add(food);
	}
	
	public void printMenu() {
		System.out.printf("***********************%s Menu***************\n", getName());
		System.out.print("\n");
		
		System.out.println("**************Drinks****************");
		for(int i = 0; i < this.drinks.size(); i++) {
			System.out.println(this.drinks.get(i).getName());
		}
		
		System.out.println("**************Entrees****************");
		for(int i = 0; i < this.meals.size(); i++) {
			System.out.println(this.meals.get(i).getName());
		}
		
		System.out.println("**************Sides****************");
		for(int i = 0; i < this.extras.size(); i++) {
			System.out.println(this.extras.get(i).getName());
		}
		
		System.out.println("**************Desserts****************");
		for(int i = 0; i < this.sweets.size(); i++) {
			System.out.println(this.sweets.get(i).getName());
		}
		
	}
	
	public void printOrder() {
		System.out.println("This is your order");
		for(int i = 0; i < this.order.size(); i++) {
			System.out.printf("%s  $%f\n",this.order.get(i).getName(), this.order.get(i).getPrice());
		}
		
	}
	
	public double sumOrder() {
		sum = 0;
		for(int i = 0; i < this.order.size(); i++) {
			sum = sum + this.order.get(i).getPrice();
		}
		return sum;
	}
}

File: Restaurant/Python/Restaurant.py
from Item import Item
from Drink import Drink
from Entree import Entree
from Side import Side
from Dessert import Dessert

class Restaurant:
	def __init__(self, name):
		self.name = name
		self.total = 0
		self.drinks = []
		self.meals = []
		self.extras = []
		self.sweets = []
		self.order = []
		
	def addDrinks(self,Drink):
		self.drinks.append(Drink)
		
	def addMeals(self,Entree):
		self.meals.append(Entree)
		
	def addSides(self, Side):
		self.extras.append(Side)
		
	def addDessert(self,Dessert):
		self.sweets.append(Dessert)
		
	def addToOrder(self, Item):
		self.order.append(Item)
		
	def printMenu(self):
		print "****************",self.name,"Menu*********************\n"
		print("\n")
		
		print("****************Drinks*********************\n")
		for x in range(len(self.drinks)):
			print self.drinks[x].getName()
		print("\n")
		
		print("****************Entrees*********************\n")
		for x in range(len(self.meals)):
			print self.meals[x].getName()
		print("\n")
		
		print("****************Sides*********************\n")
		for x in range(len(self.extras)):
			print self.extras[x].getName()
		print("\n")
		
		print("****************Dessert*********************\n")
		for x in range(len(self.sweets)):
			print self.sweets[x].getName()
		print("\n")
		
	def printOrder(self):
		print "This is your order:\n"
		for x in range(len(self.order)):
			print self.order[x].getName()
		print "\n"
		
	def sumOrder(self):
		total = 0
		for x in range(len(self.order)):
			total = total + self.order[x].getPrice()
		return total
			
	def clearOrder(self):
		del self.order[:]
	
			
		
		
		
		
		
		
		
	
		
	
